controllers/controllers/resource/template.go

```diff
@@ -71,16 +75,23 @@ func (r *VsphereTemplate) TemplateResources(ctx context.Context, eksaCluster *an
 		controlPlaneTemplateName = cp.Spec.InfrastructureTemplate.Name
 	}
 
-	var workloadTemplateName string
-	updateWorkloadTemplate := vsphere.AnyImmutableFieldChanged(oldVdc, &vdc, oldWorkerVmc, &workerVmc)
-	if updateWorkloadTemplate {
-		workloadTemplateName = templateBuilder.WorkerMachineTemplateName(clusterName)
-	} else {
-		mcDeployment, err := r.MachineDeployment(ctx, eksaCluster)
-		if err != nil {
-			return nil, err
+	var workloadTemplateNames []string
+	for _, vmc := range workerVmc {
+		oldVmc := oldWorkerVmcs[vmc.Name]
+		updateWorkloadTemplate := vsphere.AnyImmutableFieldChanged(oldVdc, &vdc, &oldVmc, vmc)
+		if updateWorkloadTemplate {
+			workloadTemplateName := templateBuilder.WorkerMachineTemplateName(clusterName, clusterSpec.Spec.WorkerNodeGroupConfigurations[0].Name)
+			workloadTemplateNames = append(workloadTemplateNames, workloadTemplateName)
+		} else {
+			mcDeployments, err := r.MachineDeployments(ctx, eksaCluster)
```

Comment:
Wondering if we should be retrieving based on name here

Reply:
I updated it make this a map & grab the correct deployment by name from there